internal/guest/runtime/hcsv2/uvm.go

```diff
@@ -937,6 +939,23 @@ func newInvalidRequestTypeError(rt guestrequest.RequestType) error {
 	return errors.Errorf("the RequestType %q is not supported", rt)
 }
 
+func modifySCSIDevice(
+	ctx context.Context,
+	rt guestrequest.RequestType,
+	msd *guestresource.SCSIDevice,
+) error {
+	switch rt {
+	case guestrequest.RequestTypeRemove:
+		cNum, err := scsi.ActualControllerNumber(ctx, msd.Controller)
+		if err != nil {
+			return err
+		}
+		return scsi.UnplugDevice(ctx, cNum, msd.Lun)
```

Comment:
what happens if we still have mounted partitions and we unplug SCSI? I think this is important in context of enforcing device unmount policy and something that we'll need to take care as part of some other PR in the future.

Reply:
Discussed offline. We agree in the future the guest should track mounts and their use, but for now there should be no negative impact to this change.